### ISSUE TITLE
ArcFaceLoss parameters become nan after several epochs 

### ISSUE THREAD

**Issue by fogside** [KevinMusgrave/pytorch-metric-learning#209]
Hi!
I'm using ArcFaceLoss with PyTorch Lightning and making all computations in fp16.
I use the same optimizer for loss and models' parameters. 
And I have a problem that after several steps of training `self.W` in loss object becomes a matrix of nan after some `-inf` values start to appear in it.
Maybe you can suggest how to solve this problem.

**Comment by KevinMusgrave** [comment]
What version of pytorch and pytorch-metric-learning are you using?
```
torch.__version__
pytorch_metric_learning.__version__
```

**Comment by fogside** [comment]
```
pytorch_metric_learning.__version__
'0.9.92'
torch.__version__
'1.6.0'
```

**Comment by KevinMusgrave** [comment]
Those versions should work. Have you tried lowering the learning rate for ArcFaceLoss?

**Comment by fogside** [comment]
Thank you!
Yes, it looks like Adam with the same initial learning rate as SGD produces nans.
With SGD optimizer it doesn't produce nans. I set my lr 0. 000001.
